### DISCUSSION TITLE
binary target support for SPM or xcframework artifact

### DISCUSSION THREAD

**Discussion by lazar-otasevic-cif** [pointfreeco/swift-snapshot-testing#648] [Ideas]
It would be nice to have XCFramework distribution, or at least xcframework artifact in releases so we don't have to rebuild the framework from source everytime we need it

**Comment by MagFer** [comment]
That would be really nice. We were thinking about replacing https://github.com/uber/ios-snapshot-test-case with this Snapshot Test framework to support snapshot tests in SwiftUI. However, our project only uses Carthage since it's supper inconvenient to refetch and rebuild the dependencies every time we check out a new commit/branch.

@lazar-otasevic-cif, do you know if it's possible to distribute binaries or precompile open source frameworks with SPM? 
Another option, as you mention, is to create an precompiled XCFramework for every release and drag and drop it, like Firebase (https://firebase.google.com/docs/ios/setup), where they alow us to fetch the XCFrameworks.zip and drag & drop them to the project. E. g.
- Download the [framework SDK zip](https://firebase.google.com/download/ios)

Let's ask @stephencelis about it, since the Carthage support has been removed in the latest version 1.10.0  https://github.com/pointfreeco/swift-snapshot-testing/discussions/642.

**Reply by stephencelis** [reply]
I replied [here](https://github.com/pointfreeco/swift-snapshot-testing/discussions/642#discussioncomment-3840596), but Carthage is definitely not a requirement for XCFramework support!

We're focussed on shipping a library via SPM at this time, and believe it should be possible for folks to build an XCFramework themselves, and we believe it could even be taken on as a third-party CI process. While we'd consider PRs to build/bundle XCFrameworks with each GitHub release, we'd have to caveat that it's not a priority for us, so if it were to break in the future, we may have to remove support.
